### PR TITLE
Adds threadPool implementation for persisting Point3D data #MOV-252

### PR DESCRIPTION
**Pull request by hb0**

### CODE REVIEW

File: datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java [hunk 1]
```diff
@@ -165,7 +166,11 @@ public void onLocationChanged(final Location location) {
 
             synchronized (this) {
                 for (CapturingProcessListener listener : this.listener) {
-                    listener.onLocationCaptured(new GeoLocation(latitude, longitude, gpsTime, speed, gpsAccuracy));
+                    try {
+                        listener.onLocationCaptured(new GeoLocation(latitude, longitude, gpsTime, speed, gpsAccuracy));
+                    } catch (DataCapturingException e) {
+                        throw new IllegalStateException(e);
```
Comment (hb0) [inline datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java]: IllegalStateException is thrown when location data is captured without GpsFix (i.e. timestamp = 0L), that should be legit, right?

File: datacapturing/src/main/java/de/cyface/datacapturing/persistence/CapturedDataWriter.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                }
+                for (int startIndex = 0; startIndex < values.length; startIndex += MAX_SIMULTANEOUS_OPERATIONS) {
+                    int endIndex = Math.min(values.length, startIndex + MAX_SIMULTANEOUS_OPERATIONS);
+                    // BulkInsert is about 80 times faster than insertBatch
+                    client.bulkInsert(MeasuringPointsContentProvider.MAGNETIC_VALUE_POINTS_URI,
+                            Arrays.copyOfRange(values, startIndex, endIndex));
+                }
+
+            } catch (DataCapturingException e) {
+                // TODO: not sure if this is ok according to our SDK API?
```
Comment (hb0) [inline datacapturing/src/main/java/de/cyface/datacapturing/persistence/CapturedDataWriter.java]: If client == null it should be legit to throw an IllegalStateException, can this actually happen when used properly?

File: datacapturing/src/main/java/de/cyface/datacapturing/persistence/MeasurementPersistence.java [hunk 3]
```diff
@@ -197,7 +113,13 @@ public void storeData(final @NonNull CapturedData data, final long measurementId
      * @param location The geo location to store.
      * @param measurementIdentifier The identifier of the measurement to store the data to.
      */
-    public void storeLocation(final @NonNull GeoLocation location, final long measurementIdentifier) {
+    public void storeLocation(final @NonNull GeoLocation location, final long measurementIdentifier)
+            throws DataCapturingException {
+        if (location.getTimestamp() == 0L) {
+            throw new DataCapturingException(
+                    "GeoLocations captured without GpsFix. Make sure you only capture data when a fix is available.");
```
Comment (hb0) [inline datacapturing/src/main/java/de/cyface/datacapturing/persistence/MeasurementPersistence.java]: This came from the 1.0.0 code so I migrated it back here as this should still be the case (timestamp 0L if no GpsFix) and such location data should never be stored, right?

File: datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java (hunk 1 above)
Comment (muthenberg) [reply datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java]: It is legit for framework functions, which you cannot change to throw a DataCapturingException. For all other cases use a typed exception.

**Pull request merged**